ext/standard/array.c

```diff
@@ -3372,6 +3375,15 @@ static void php_splice(HashTable *in_hash, zend_long offset, zend_long length, H
 	HT_SET_ITERATORS_COUNT(&out_hash, HT_ITERATORS_COUNT(in_hash));
 	HT_SET_ITERATORS_COUNT(in_hash, 0);
 	in_hash->pDestructor = NULL;
+	
+	if (UNEXPECTED(GC_DELREF(in_hash) == 0)) {
+		/* Array was completely deallocated during the operation */
+		zend_array_destroy(in_hash);
+		zend_hash_destroy(&out_hash);
+		zend_throw_error(NULL, "Array was modified during array_splice operation");
+		return;
```

Comment:
`RETURN_THROWS();` is the idiomatic way to do this.

Reply:
Ah, I missed this is not the function itself. Looks good then!

Reply:
Otherwise, I get this compile error:

```
/Users/alex/GolandProjects/php-src/ext/standard/array.c:3384:3: error: use of undeclared identifier 'return_value'
   3384 |                 RETURN_THROWS();
        |                 ^
  /Users/alex/GolandProjects/php-src/Zend/zend_API.h:1053:69: note: expanded from macro 'RETURN_THROWS'                                                                    
              
   1053 | #define RETURN_THROWS()                                 do { ZEND_ASSERT(EG(exception)); (void) return_value; return; } while (0)"
```